solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -84,6 +84,11 @@ public void init(List<String> solrUrls) throws Exception {
   /** Create a SolrClient implementation that uses the specified Solr node URL */
   protected abstract SolrClient getSolrClient(String baseUrl);
 
+  @Override
```

Comment:
Super class ClusterStateProvider's implementation calls getClusterState().getCollectionOrNull(collectionName) - in that case BaseHttpCSP could make a call to fetch the entire cluster state! This change is to prevent that from happening.

Reply:
Done